plugins/Goals/Columns/Metrics/RevenuePerVisit.php

```diff
@@ -50,14 +50,15 @@ public function compute(Row $row)
 
         $revenue = 0;
         foreach ($goals as $goalId => $goalMetrics) {
-            if ($goalId == Piwik::LABEL_ID_GOAL_IS_ECOMMERCE_CART) {
+            if (is_numeric($goalId) && $goalId <= GoalManager::IDGOAL_ORDER) {
```

Comment:
Wouldn't `$goalId <= GoalManager::IDGOAL_ORDER` cause ecommerce order goals to be skipped?

eg. if `$goalId` is 0 (an ecommerce order) then it is equal to `GoalManager::IDGOAL_ORDER` and the condition will be true.

```suggestion
            if (is_numeric($goalId) && $goalId < GoalManager::IDGOAL_ORDER) {
```

Reply:
Based on what the code was doing when it was created (on PHP 5) it would be correct to exclude orders here. But reading the code and considering what the metric should represent, I would say we should include goals here.
So basically this was always broken on PHP 7, but PHP 8 magically fixed it already 🙈